File: wash_away/lib/models/address_model.dart
// --- Address Data Model ---
class Address {
  final String? id;
  final String label; // e.g., Home, Office
  final String fullAddress;
  final double? latitude;
  final double? longitude;
  bool isDefault;

  Address({
    this.id,
    required this.label,
    required this.fullAddress,
    this.latitude,
    this.longitude,
    this.isDefault = false,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['_id']?.toString() ?? json['id']?.toString(),
      label: json['label'] ?? '',
      fullAddress: json['full_address'] ?? '',
      latitude: json['latitude']?.toDouble(),
      longitude: json['longitude']?.toDouble(),
      isDefault: json['is_default'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) '_id': id,
      'label': label,
      'full_address': fullAddress,
      if (latitude != null) 'latitude': latitude,
      if (longitude != null) 'longitude': longitude,
      'is_default': isDefault,
    };
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Address &&
          runtimeType == other.runtimeType &&
          id == other.id;

  @override
  int get hashCode => id.hashCode;
}

File: backend/src/models/Address.model.js
import mongoose from 'mongoose';

const addressSchema = new mongoose.Schema({
  customer_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  label: {
    type: String,
    enum: ['home', 'office', 'other'],
    default: 'other'
  },
  address_line: {
    type: String,
    required: true,
    trim: true
  },
  city: {
    type: String,
    trim: true
  },
  state: {
    type: String,
    trim: true
  },
  zip_code: {
    type: String,
    trim: true
  },
  country: {
    type: String,
    trim: true,
    default: 'USA'
  },
  latitude: {
    type: Number
  },
  longitude: {
    type: Number
  },
  is_default: {
    type: Boolean,
    default: false
  },
  created_date: {
    type: Date,
    default: Date.now
  },
  updated_date: {
    type: Date,
    default: Date.now
  }
}, {
  timestamps: { createdAt: 'created_date', updatedAt: 'updated_date' }
});

// Index for faster queries
addressSchema.index({ customer_id: 1 });
addressSchema.index({ customer_id: 1, is_default: 1 });

const Address = mongoose.model('Address', addressSchema);

export default Address;



